From 4ea937a55caaa1f28da25644abe9ae58b9176ee7 Mon Sep 17 00:00:00 2001
From: Philipp Hancke <phancke@meta.com>
Date: Thu, 3 Jul 2025 07:35:32 -0700
Subject: [PATCH 2/4] IWYU modules/video_capture

using
  find modules/video_capture -name "*.h" -o -name "*.cc" | xargs tools_webrtc/iwyu/apply-include-cleaner
followed by
  tools_webrtc/gn_check_autofix.py -C out/Default
followed by
  git cl format
followed by manual replacement of C style headers with their C++
equivalents and moving around includes.

Bug: webrtc:42226242
Change-Id: If7f5f496452eb0dcf4a84ffe37c77e4e471b9f38
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/399061
Reviewed-by: Ilya Nikolaevskiy <ilnik@webrtc.org>
Commit-Queue: Philipp Hancke <phancke@meta.com>
Reviewed-by: Sergey Silkin <ssilkin@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#45130}
---
 modules/video_capture/linux/pipewire_session.cc | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
index 990bfde912..2c533a0edb 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
@@ -17,13 +17,23 @@
 #include <spa/pod/parser.h>
 
 #include <algorithm>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <optional>
 
+#include "absl/strings/string_view.h"
 #include "common_video/libyuv/include/webrtc_libyuv.h"
-#include "modules/video_capture/device_info_impl.h"
+#include "modules/portal/pipewire_utils.h"
+#include "modules/portal/portal_request_response.h"
+#include "modules/video_capture/linux/camera_portal.h"
+#include "modules/video_capture/video_capture_defines.h"
+#include "modules/video_capture/video_capture_options.h"
 #include "rtc_base/logging.h"
 #include "rtc_base/sanitizer.h"
-#include "rtc_base/string_encode.h"
 #include "rtc_base/string_to_number.h"
+#include "rtc_base/synchronization/mutex.h"
 
 namespace webrtc {
 namespace videocapturemodule {
-- 
2.51.0

